### PR TITLE
feat: support auto installing terraform binary

### LINKED ISSUE #769: Feat: Automatically download Terraform binary if it does not exist
### What would you like to be added?

Automatically download Terraform binary if it does not exist

### Why is this needed?

Reduce prerequisite steps and user's cognitive load

### PR DESCRIPTION
**Pull request by liu-hm19**
#### What type of PR is this?

/kind feature
#### What this PR does / why we need it:
This PR supports automatically installing terraform binary and setting the environment variable of executable path before `kusion preview`, `kusion apply` and `kusion destroy`, if terraform has not been downloaded before. 

Here is an example of executing `kusion preview` twice on a machine where the terraform executable binary has not yet been downloaded. 
image
image

#### Which issue(s) this PR fixes:

Fixes #769 

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., design docs, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/engine/runtime/terraform/util.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	"kusionstack.io/kusion/pkg/util/kfile"
+	"kusionstack.io/kusion/pkg/util/pretty"
+)
+
+var (
+	tfInstallSubDir  = "terraform"
+	tfInstallTimeout = 3 * time.Minute
+)
+
+type InstallOption struct {
```
Comment (adohe) [inline pkg/engine/runtime/terraform/util.go:24]: I would suggest rename `InstallOption` to `CLIInstaller`, also rename `util.go` to `installer.go`

File: pkg/engine/runtime/terraform/util.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+		if !io.MutePrompt {
+			sp.WarningPrinter.Printfln("Terraform executable binary is not found")
+		}
+
+		if err := io.installTerraform(); err != nil {
+			return err
+		}
+
+		if !io.MutePrompt {
+			sp.SuccessPrinter.Printfln("Successfully installed terraform and set the executable path")
```
Comment (adohe) [inline pkg/engine/runtime/terraform/util.go:56]: I am not sure whether it's a good chose to print terrform related words in the console, how does radius work?

Comment (liu-hm19) [reply pkg/engine/runtime/terraform/util.go]: Radius prints relevant information in the log, and I have updated the implementation here.

**Review by adohe — APPROVED** [review]
/lgtm

**Pull request merged**